addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithNavigationControlXML.java

```diff
@@ -0,0 +1,361 @@
+                        + state.getAllItemLabels());
+                return;
+            }
+        }
+    }
+
+    /**
+     * Finds an item on the current page. A page contains up to 8 items.
+     * Operates on a cached XML node! Call refreshMenuState for up-to-date information.
+     *
+     * @return Return the item index [1,8] or -1 if not found.
+     */
+    private int findItemOnCurrentPage(String itemName) {
+        for (int i = 0; i < MAX_PER_PAGE; i++) {
+            if (itemName.equals(state.items[i])) {
+                return i + 1;
+            }
+        }
+
+        return -1;
+    }
+
+    private boolean selectItem(String name) throws IOException, ReceivedMessageParseException {
+        final int pageCount = (int) Math.ceil(state.maxLine / (double) MAX_PER_PAGE);
+        final int currentPage = (int) Math.floor((state.currentLine - 1) / (double) MAX_PER_PAGE);
+
+        AbstractConnection com = comReference.get();
+        for (int pageIndex = 0; pageIndex < pageCount; pageIndex++) {
+            // Start with the current page and then go to the end and start at page 1 again
+            int realPage = (currentPage + pageIndex) % pageCount;
+
+            if (currentPage != realPage) {
+                goToPage(pageIndex);
+            }
+
+            int index = findItemOnCurrentPage(name);
+            if (index > 0) {
+                com.send(wrInput(
+                        "<List_Control><Direct_Sel>Line_" + String.valueOf(index) + "</Direct_Sel></List_Control>"));
+                update();
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Refreshes the menu state and caches the List_Info node from the response. This method may take
+     * some time because it retries the request for up to MENU_MAX_WAITING_TIME or the menu state reports
+     * "Ready", whatever comes first.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        int totalWaitingTime = 0;
+
+        Document doc;
+        Node currentMenu;
+
+        AbstractConnection com = comReference.get();
+        while (true) {
+            String response = com.sendReceive(wrInput("<List_Info>GetParam</List_Info>"));
+            doc = XMLUtils.xml(response);
+            if (doc.getFirstChild() == null) {
+                throw new ReceivedMessageParseException("<List_Info>GetParam failed: " + response);
+            }
+
+            currentMenu = XMLUtils.getNode(doc.getFirstChild(), "List_Info");
+
+            if (currentMenu == null) {
+                throw new ReceivedMessageParseException("<List_Info>: GetParam response invalid!");
+            }
+
+            Node nodeMenuState = XMLUtils.getNode(currentMenu, "Menu_Status");
+
+            if (nodeMenuState == null || nodeMenuState.getTextContent().equals("Ready")) {
```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)